common/service.go

```diff
@@ -151,7 +151,7 @@ func (bs *BaseService) Reset() (bool, error) {
 		return false, nil
 	}
 	// never happens
-	return false, nil
+	return false, nil // nolint: vet
```

Comment:
whats the problem here?

Reply:
see comment on L153 ... never happens, so go vet picks up on it

Reply:
@zramsay perhaps instead of adding the nolint clause we can actually fix the code
```diff
diff --git a/common/service.go b/common/service.go
index 71fc03c..dc321e5 100644
--- a/common/service.go
+++ b/common/service.go
@@ -146,11 +146,8 @@ func (bs *BaseService) Reset() (bool, error) {
 
 		bs.Quit = make(chan struct{})
 		return true, bs.impl.OnReset()
-	} else {
-		bs.Logger.Debug(Fmt("Can't reset %v. Not stopped", bs.name), "impl", bs.impl)
-		return false, nil
 	}
-	// never happens
+	bs.Logger.Debug(Fmt("Can't reset %v. Not stopped", bs.name), "impl", bs.impl)
 	return false, nil
 }
 ```

or for easier readability
```diff
diff --git a/common/service.go b/common/service.go
index 71fc03c..2d86baa 100644
--- a/common/service.go
+++ b/common/service.go
@@ -140,18 +140,16 @@ func (bs *BaseService) OnStop() {}
 
 // Implements Service
 func (bs *BaseService) Reset() (bool, error) {
-	if atomic.CompareAndSwapUint32(&bs.stopped, 1, 0) {
-		// whether or not we've started, we can reset
-		atomic.CompareAndSwapUint32(&bs.started, 1, 0)
-
-		bs.Quit = make(chan struct{})
-		return true, bs.impl.OnReset()
-	} else {
+	if stopped := atomic.CompareAndSwapUint32(&bs.stopped, 1, 0); !stopped {
 		bs.Logger.Debug(Fmt("Can't reset %v. Not stopped", bs.name), "impl", bs.impl)
 		return false, nil
 	}
-	// never happens
-	return false, nil
+
+	// whether or not we've started, we can reset
+	atomic.CompareAndSwapUint32(&bs.started, 1, 0)
+
+	bs.Quit = make(chan struct{})
+	return true, bs.impl.OnReset()
 }
 
 // Implements Service
```